test/Graphics/Transform.cpp

```diff
@@ -0,0 +1,353 @@
+#include <SFML/Graphics/Transform.hpp>
+#include "GraphicsUtil.hpp"
+#include "SystemUtil.hpp"
+
+#include <doctest.h>
+
+using doctest::Approx;
+
+TEST_CASE("sf::Transform class - [graphics]")
+{
+    SUBCASE("Construction")
+    {
+        SUBCASE("Default constructor")
+        {
+            const sf::Transform transform;
+            CHECK(transform == sf::Transform::Identity);
+            CHECK(transform.getMatrix()[0] == 1.0f);
+            CHECK(transform.getMatrix()[1] == 0.0f);
+            CHECK(transform.getMatrix()[2] == 0.0f);
+            CHECK(transform.getMatrix()[3] == 0.0f);
+            CHECK(transform.getMatrix()[4] == 0.0f);
+            CHECK(transform.getMatrix()[5] == 1.0f);
+            CHECK(transform.getMatrix()[6] == 0.0f);
+            CHECK(transform.getMatrix()[7] == 0.0f);
+            CHECK(transform.getMatrix()[8] == 0.0f);
+            CHECK(transform.getMatrix()[9] == 0.0f);
+            CHECK(transform.getMatrix()[10] == 1.0f);
+            CHECK(transform.getMatrix()[11] == 0.0f);
+            CHECK(transform.getMatrix()[12] == 0.0f);
+            CHECK(transform.getMatrix()[13] == 0.0f);
+            CHECK(transform.getMatrix()[14] == 0.0f);
+            CHECK(transform.getMatrix()[15] == 1.0f);
+        }
+
+        SUBCASE("3x3 matrix constructor")
+        {
+            const sf::Transform transform(3.14f, 3.14f, 3.14f,
+                                          3.14f, 3.14f, 3.14f,
+                                          3.14f, 3.14f, 3.14f);
```

Comment:
Wouldn't it be better to have distinct components?
This test can pass by having those values in the wrong places.

Reply:
You're right. I changed this so each value is unique.